File: 2025-08-31/hex_generator.py
def generate_hex(color):
    output_color = ""
    max_possible_value = 255

    color_object = {
    'red_component':   {'is_dominant': None, 'hex_code': None},
    'green_component': {'is_dominant': None, 'hex_code': None},
    'blue_component':  {'is_dominant': None, 'hex_code': None}
  }

    if color == "red" or color == "green" or color == "blue":
        color_object[color + '_component']['is_dominant'] = True
    else:
        return "Invalid color"

    for color_component in color_object:
        if color_object[color_component]['is_dominant']:
            color_object[color_component]['hex_code'] = generate_random_color_code(True, max_possible_value)
            max_possible_value = int(color_object[color_component]['hex_code'], 16) - 1
            break

    for color_component in color_object:
        if not color_object[color_component]['is_dominant']:
            color_object[color_component]['hex_code'] = generate_random_color_code(False, max_possible_value)
        output_color += color_object[color_component]['hex_code']

    return output_color

def generate_random_color_code(is_dominant, max_value):
    import random

    min_value = 0
    if is_dominant:
        min_value = 1

    return f"{random.randint(min_value, max_value):02X}"

print(generate_hex("yellow"))
print(generate_hex("red"))
print(generate_hex("green"))
print(generate_hex("blue"))

File: 2025-08/2025-08-31/hexGenerator.js
function generateHex(color) {
  let outputColor = "";
  let maxPossibleValue = 256;

  let colorObject = {
    redComponent:   {isDominant: null, hexCode: null},
    greenComponent: {isDominant: null, hexCode: null},
    blueComponent:  {isDominant: null, hexCode: null}
  };

  if (color == "red" || color == "green" || color == "blue") 
    colorObject[color + "Component"].isDominant = true;
  else 
    return "Invalid color";

  for (let colorComponent in colorObject) {
    if (colorObject[colorComponent].isDominant) {
      colorObject[colorComponent].hexCode = generateRandomColorCode(true, maxPossibleValue);
      maxPossibleValue = parseInt(colorObject[colorComponent].hexCode, 16);
      break;
    }
  }

  for (let colorComponent in colorObject) {
    if (!colorObject[colorComponent].isDominant) {
      colorObject[colorComponent].hexCode = generateRandomColorCode(false, maxPossibleValue);
    }
    outputColor += colorObject[colorComponent].hexCode;
  }

  return outputColor;
}

function generateRandomColorCode(isDominant, maxValue) {
  let minValue = 0;
  if (isDominant) {
    minValue = 1;
  }
  return Math.floor(Math.random() * (maxValue - minValue) + minValue).toString(16).toUpperCase().padStart(2, "0");
}

console.log(generateHex("yellow"));
console.log(generateHex("red"));
console.log(generateHex("green"));
console.log(generateHex("blue"));